src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurCashAdvancePdpFeedFileServiceImpl.java

```diff
@@ -1,36 +1,299 @@
+            }
+        }
+        return pdpFileSuccessfullyCreated;
+    }
+
+    public void createDoneFileForPdpFile(String fullyQualifiedPdpFileName) throws IOException {
+        String fullFilePath = StringUtils.replace(fullyQualifiedPdpFileName, ConcurConstants.XML_FILE_EXTENSION, GeneralLedgerConstants.BatchFileSystem.DONE_FILE_EXTENSION);
+        FileUtils.touch(new File(fullFilePath));
+    }
+
+    public void setPaymentImportDirectory(String paymentImportDirectory) {
+        this.paymentImportDirectory = paymentImportDirectory;
+    }
+
+    public String getPaymentImportDirectory() {
+        return this.paymentImportDirectory;
+    }
+
+    public ParameterService getParameterService() {
+        return parameterService;
+    }
+
+    public void setParameterService(ParameterService parameterService) {
+        this.parameterService = parameterService;
+    }
+
+    public DataDictionaryService getDataDictionaryService() {
+        return dataDictionaryService;
+    }
+
+    public void setDataDictionaryService(DataDictionaryService dataDictionaryService) {
+        this.dataDictionaryService = dataDictionaryService;
+    }
+
+    public ConcurRequestedCashAdvanceService getConcurRequestedCashAdvanceService() {
+        return concurRequestedCashAdvanceService;
+    }
+
+    public void setConcurRequestedCashAdvanceService(ConcurRequestedCashAdvanceService concurRequestedCashAdvanceService) {
+        this.concurRequestedCashAdvanceService = concurRequestedCashAdvanceService;
+    }
+
+    public CUMarshalService getCuMarshalService() {
+        return cuMarshalService;
+    }
+
+    public void setCuMarshalService(CUMarshalService cuMarshalService) {
+        this.cuMarshalService = cuMarshalService;
+    }
+
+    public DateTimeService getDateTimeService() {
+        return DateTimeService;
+    }
+
+    public void setDateTimeService(DateTimeService dateTimeService) {
+        DateTimeService = dateTimeService;
+    }
+
+    public int getPayeeNameFieldSize() {
+        if (payeeNameFieldSize == 0) {
+            payeeNameFieldSize = retrievePayeeNameFieldSize();
+        }
+        return payeeNameFieldSize;
+    }
+
+    private PdpFeedFileBaseEntry buildPdpFeedBaseEntry(ConcurRequestExtractFile requestExtractFile) {
+        int totalPdpDetailRecordsCount = 0;
+        KualiDecimal totalPdpDetailRecordsAmount = KualiDecimal.ZERO;
+        PdpFeedFileBaseEntry pdpBaseEntry = new PdpFeedFileBaseEntry();
+        pdpBaseEntry.setHeader(buildPdpFeedHeaderEntry(requestExtractFile.getBatchDate()));
+        List<PdpFeedGroupEntry> groupEntries = new ArrayList<PdpFeedGroupEntry>();
+        for (ConcurRequestExtractRequestDetailFileLine detailFileLine : requestExtractFile.getRequestDetails()) {
+            if (detailFileLine.getValidationResult().isCashAdvanceLine() && detailFileLine.getValidationResult().isValidCashAdvanceLine()) {
```

Comment:
At this point, I am not going to making that assumption. Having both checks makes the code and logic clearer IMHO.  I only want to process the valid cash advance lines, or only output the data to the log for invalid cash advance lines, and skip any lines that are not cash advances.  With the data I have encountered, I'd prefer to leave both checks in.